File: src/Transcript.java
import java.util.ArrayList;

public class Transcript {
    private double GPA_100;
    private int term;
    private int totalCredit; //
    private ArrayList<GradeClass> passedCourses, failedCourses;

    public Transcript(double GPA_100, int term, ArrayList<GradeClass> passedCourses, ArrayList<GradeClass> failedCourses) {
        this.GPA_100 = GPA_100;
        this.term = term;
        totalCredit = 0;
        this.passedCourses = passedCourses;
        this.failedCourses = failedCourses;
        this.totalCredit = calculateTotalCredit(); //*

    }

    public int calculateTotalCredit() { //*
        for(int i = 0 ; i < passedCourses.size() ; i++) {
            totalCredit += passedCourses.get(i).getCourse().getCredit(); //
        }
        return totalCredit; //*
    }

    public double getGPA_100() {
        return GPA_100;
    }

    public void setGPA_100(double GPA_100) {
        this.GPA_100 = GPA_100;
    }

    public int getTerm() {
        return term;
    }

    public void setYear(int term) {
        this.term = term;
    }

    public ArrayList<GradeClass> getPassedCourses() {
        return passedCourses;
    }

    public ArrayList<GradeClass> getFailedCourses() {
        return failedCourses;
    }

    public int getTotalCredit() { //
        return totalCredit;
    }

    public void setTotalCredit(int totalCredit) { //
        this.totalCredit = totalCredit;
    }
}

File: iter3/python_development/CourseObject/Transcript.py
class Transcript:
    def __init__(self, GPA_100, term, passedCourses, failedCourses):
        self.GPA_100 = GPA_100
        self.term = term
        self.totalCredit = 0
        self.passedCourses = passedCourses
        self.failedCourses = failedCourses
        self.totalCredit = self.calculateTotalCredit()

    def calculateTotalCredit(self):
        for gradeClass in self.passedCourses:
            self.totalCredit += gradeClass.getCourse().getCredit()
        return self.totalCredit

    def getGPA_100(self):
        return self.GPA_100

    def setGPA_100(self, GPA_100):
        self.GPA_100 = GPA_100

    def getTerm(self):
        return self.term

    def setTerm(self, term):
        self.term = term

    def getPassedCourses(self):
        return self.passedCourses

    def getFailedCourses(self):
        return self.failedCourses

    def getTotalCredit(self):
        return self.totalCredit

    def setTotalCredit(self, totalCredit):
        self.totalCredit = totalCredit
